modules/tooling-support-parent/tooling-support/src/main/java/org/mule/runtime/module/tooling/internal/config/InternalDeclarationSession.java

```diff
@@ -100,215 +63,129 @@
   @Inject
   private ExpressionManager expressionManager;
 
-  private LazyValue<ArtifactHelper> artifactHelperLazyValue;
+  private final LazyValue<ArtifactHelper> artifactHelperLazyValue;
+  private final LazyValue<ValueProviderExecutor> valueProviderExecutorLazyValue;
+  private final LazyValue<MetadataKeysExecutor> metadataKeysExecutorLazyValue;
+  private final LazyValue<MetadataComponentExecutor> metadataComponentExecutorLazyValue;
 
   InternalDeclarationSession(ArtifactDeclaration artifactDeclaration) {
     this.artifactHelperLazyValue =
         new LazyValue<>(() -> new ArtifactHelper(extensionManager, componentLocator, artifactDeclaration));
+    this.valueProviderExecutorLazyValue =
+        new LazyValue<>(() -> new ValueProviderExecutor(muleContext, connectionManager, expressionManager, reflectionCache,
+                                                        artifactHelper()));
+    this.metadataKeysExecutorLazyValue =
+        new LazyValue<>(() -> new MetadataKeysExecutor(connectionManager, reflectionCache, artifactHelper()));
+    this.metadataComponentExecutorLazyValue =
+        new LazyValue<>(() -> new MetadataComponentExecutor(connectionManager, reflectionCache, artifactHelper()));
```

Comment:
artifactHelper is  being initialized once any of these 3 lazyValues are. That should not be a problem but maybe it would be better for them to accept LazyValue<ArtifactHelper> ?

Reply:
Actually the artifactHelper is used before to validate extension model from declaration so at the point the executor is used the artifactHelper is already initialized.